concordium-contracts-common/src/schema.rs

```diff
@@ -780,6 +1081,192 @@ impl Deserial for FunctionV2 {
     }
 }
 
+impl Serial for FunctionV3 {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        match self {
+            FunctionV3::Param(parameter) => {
+                out.write_u8(0)?;
+                parameter.serial(out)
+            }
+            FunctionV3::Rv(return_value) => {
+                out.write_u8(1)?;
+                return_value.serial(out)
+            }
+            FunctionV3::ParamRv {
+                parameter,
+                return_value,
+            } => {
```

Comment:
If you use the struct, as I suggest you do, then you can use a single `u8` tag for encoding which parameters are `Some` (e.g., bit 0 means parameter is Some, bit 1 set means return value is Some, ...)